vault/identity_store_oidc.go

```diff
@@ -1193,18 +1219,34 @@ func (i *IdentityStore) pathOIDCReadPublicKeys(ctx context.Context, req *logical
 		return nil, err
 	}
 	if len(keys) > 0 {
-		v, ok, err := i.oidcCache.Get(noNamespace, "nextRun")
+		// if maxJwksClientCache is set use that, otherwise fall back on the more conservative
+		// nextRun values
+		maxJwksClientCache, ok, err := i.oidcCache.Get(noNamespace, "maxJwksClientCache")
 		if err != nil {
 			return nil, err
 		}
 
 		if ok {
-			now := time.Now()
-			expireAt := v.(time.Time)
-			if expireAt.After(now) {
-				expireInSeconds := expireAt.Sub(time.Now()).Seconds()
-				expireInString := fmt.Sprintf("max-age=%.0f", expireInSeconds)
-				resp.Data[logical.HTTPRawCacheControl] = expireInString
+			maxDuration := int64(maxJwksClientCache.(time.Duration))
+			randDuration := mathrand.Int63n(maxDuration)
+			// truncate to seconds
+			durationInSeconds := time.Duration(randDuration).Seconds()
+			durationInString := fmt.Sprintf("max-age=%.0f", durationInSeconds)
+			resp.Data[logical.HTTPRawCacheControl] = durationInString
+		} else {
+			v, ok, err := i.oidcCache.Get(noNamespace, "nextRun")
+			if err != nil {
+				return nil, err
+			}
+
+			if ok {
+				now := time.Now()
+				expireAt := v.(time.Time)
+				if expireAt.After(now) {
+					expireInSeconds := expireAt.Sub(time.Now()).Seconds()
+					expireInString := fmt.Sprintf("max-age=%.0f", expireInSeconds)
+					resp.Data[logical.HTTPRawCacheControl] = expireInString
+				}
```

Comment:
Nit: Can we break this out into its own function? There are quite a few layers of indenting on that can be avoided by early returns.